### PR TITLE
Add more info when register with RHC

### PR DESCRIPTION
**Pull request by bzwei**
In the log:
```
{"level":"info","msg":"Catalog Worker Version 0.0.1 GIT SHA e011bbc638b3b3c0c35796e961d91be2444e42c0 Build 2021-02-16 13:51","time":"2021-02-16T13:51:10-05:00"}
```

### CODE REVIEW

File: internal/common/common.go [hunk 1]
```diff
@@ -2,6 +2,17 @@ package common
 
 import "time"
 
+const (
```
Comment (mkanoor) [inline internal/common/common.go]: @bzwei Wouldn't this have to change
https://github.com/RedHatInsights/rhc-worker-catalog/blob/367eee14b8dce7db9fbee0dc27863c2207937d5c/Makefile#L9

Can you run and log the line to see what it shows

File: internal/request/grpc.go [hunk 2]
```diff
@@ -76,7 +78,20 @@ func registerRHCWorker() string {
...
+		Handler: "catalog",
+		Pid:     int64(os.Getpid()),
+		Features: map[string]string{
+			"SourceRef":       viper.GetString("ANSIBLE_TOWER.uuid"),
+			"SrcName":         viper.GetString("ANSIBLE_TOWER.name"),
+			"SrcType":         "ansible-tower",
+			"ApplicationType": "/insights/platform/catalog",
+			"WorkerVersion":   build.Version,
+			"WorkerBuild":     build.Build,
+			"WorkerSHA":       build.Sha1,
```
Comment (mkanoor) [inline internal/request/grpc.go:91]: @bzwei Should we also add the Date when the binary was built?

File: Makefile [hunk 3]
```diff
@@ -6,7 +6,7 @@ BINARY=rhc_catalog_worker
 .DEFAULT_GOAL := build
 
 build:
-	go build -ldflags="-X 'main.Version=${VERSION}' -X main.Sha1=`git rev-parse HEAD`" -o ${BINARY} ${SRC_FILES}
+	go build -ldflags="-X 'github.com/RedHatInsights/rhc-worker-catalog/build.Version=${VERSION}' -X github.com/RedHatInsights/rhc-worker-catalog/build.Sha1=`git rev-parse HEAD`" -o ${BINARY} ${SRC_FILES}
```
Comment (mkanoor) [inline Makefile]: @bzwei Should we create variables for the VERSION, BUILD etc that are built from the output of the git commands so they are not deeply nested

File: internal/request/grpc.go (hunk 2 above)
Comment (bzwei) [reply internal/request/grpc.go:91]: I change the Build content to built date but keep the key `WorkerBuild`

File: Makefile [hunk 4]
```diff
@@ -1,12 +1,17 @@
...
 SRC_FILES= main.go
 OTHER_FILES= internal/filters/filters.go \
 	     internal/artifacts/artifacts.go
 BINARY=rhc_catalog_worker
 .DEFAULT_GOAL := build
 
+LDFLAGS :=
+LDFLAGS += -X 'github.com/RedHatInsights/rhc-worker-catalog/build.Version=${VERSION}'
+LDFLAGS += -X 'github.com/RedHatInsights/rhc-worker-catalog/build.Build=${BUILD_DATE}'
+LDFLAGS += -X 'github.com/RedHatInsights/rhc-worker-catalog/build.Sha1=`git rev-parse HEAD`'
```
Comment (mkanoor) [inline Makefile]: @bzwei Should we just add a variable for the SHA and use it here, I don't know why I added Sha1 it should have been just Sha. Also is there a way we can get the version from the tags. Add a tag to our repo.

File: build/version.go [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+	// Build is the built date for the release
+	Build string
+)
+
+func init() {
+	if Version == "" {
+		Version = "development"
+	}
+	if Sha1 == "" {
```
Comment (mkanoor) [inline build/version.go:18]: @bzwei I think this should be just SHA I think

Comment (mkanoor) [reply build/version.go:18]: @bzwei My bad it should be SHA1.

File: Makefile (hunk 4 above)
Comment (mkanoor) [reply Makefile]: @bzwei My bad it is SHA-1, so its ok.

File: Makefile [hunk 6]
```diff
@@ -1,12 +1,17 @@
 VERSION=0.0.1
+BUILD_DATE := `date +%Y-%m-%d\ %H:%M`
```
Comment (mkanoor) [inline Makefile]: @bzwei maybe include the %S for completeness

**Review by mkanoor — APPROVED** [review]

**Pull request merged**